python/semantic_kernel/semantic_functions/chat_prompt_template.py

```diff
@@ -35,29 +43,65 @@ async def render_async(self, context: "SKContext") -> str:
         )
 
     def add_system_message(self, message: str) -> None:
-        self.add_message("system", message)
+        self.add_message(ChatMessage(role="system", message=message))
 
     def add_user_message(self, message: str) -> None:
-        self.add_message("user", message)
+        self.add_message(
+            ChatMessage(
+                role="user",
+                message=message,
+            )
+        )
 
     def add_assistant_message(self, message: str) -> None:
-        self.add_message("assistant", message)
+        self.add_message(
```

Comment:
I think this would fine combined with add_assistent_message_with_function_call and have the function_call be an optional parameter. Was there specific rationale for separating the two?

Reply:
Mostly that usually only message or function_call is returned, but it can be both, which would mean setting both message and function_call as optional, and then checking if at least one is present.